great_expectations/expectations/metrics/column_aggregate_metrics/column_nonnull_count.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from great_expectations.compatibility.pyspark import functions as F
+from great_expectations.compatibility.sqlalchemy import sqlalchemy as sa
+from great_expectations.execution_engine import (
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.column_aggregate_metric_provider import (
+    ColumnAggregateMetricProvider,
+    column_aggregate_partial,
+    column_aggregate_value,
+)
+
+
+class ColumnNonNullCount(ColumnAggregateMetricProvider):
```

Comment:
You can add tests for this metric in this directory: https://github.com/great-expectations/great_expectations/tree/develop/tests/integration/metrics/column

There are other metric tests there that can serve as examples. Basically you set up a dataframe with fake data in it. Then you use the decorator `parameterize_batch_for_data_sources` on your tests which will control adding the dataframe as a table to different datasources and running your test against each.  Happy to chat sync about this.